controllers/awsmachine_controller.go

```diff
@@ -564,6 +568,20 @@ func (r *AWSMachineReconciler) reconcileLBAttachment(machineScope *scope.Machine
 	}
 
 	elbsvc := elb.NewService(clusterScope)
+
+	// In order o prevent sending request to a "not-ready" control plane machines, it is required to remove the machine
+	// from the ELB as soon as the machine gets delete or when the machine is detected or not running
+	// Nb. this is a reinforcement to the ELB health check, which blocks traffic only after a timeout is reached.
+	if !machineScope.AWSMachine.DeletionTimestamp.IsZero() || !machineScope.InstanceIsRunning() {
```

Comment:
Does this also need to filter out "Pending" as well, otherwise it looks like we might potentially de-register an instance from the ELB prior to it becoming ready on initial creation.

Reply:
InstanceIsRunning already includes AWS machine states "Pending" and "Running"

Reply:
Is elbsvc.DeregisterInstanceFromAPIServerELB(i) allowed for stopping or stopped instances?
If so, that's fine.
Otherwise might need to filter those out given this is now not called within the operational states.

Reply:
This is not clear from the API documentation, but from my tests it is allowed to call elbsvc.DeregisterInstanceFromAPIServerELB(i) for stopping or stopped instances